packages/eslint-plugin/src/rules/no-implied-eval.ts

```diff
@@ -80,13 +81,7 @@ export default createRule({
 
       // eslint-disable-next-line @typescript-eslint/no-unsafe-enum-comparison
       if (symbol && symbol.escapedName === FUNCTION_CONSTRUCTOR) {
-        const declarations = symbol.getDeclarations() ?? [];
-        for (const declaration of declarations) {
-          const sourceFile = declaration.getSourceFile();
-          if (services.program.isSourceFileDefaultLibrary(sourceFile)) {
-            return true;
-          }
-        }
+        return isFunctionLike(services.program, type);
       }
```

Comment:
`isFunctionLike` internally already checks that the type has a symbol, and that symbol is named `Function`, so the condition above is redundant.